### PR TITLE
waveshare2in13v2: Support rotation of image

### PR DESCRIPTION
**Pull request by hansmi**
A new option named `Origin` controls which of the four corners of the display should be used as the (0,0) position for the image. The included unittests may seem excessive, but they are the result of many off-by-one errors and other mistakes I made while developing this series.

### CODE REVIEW

File: waveshare2in13v2/drawing.go [hunk 1]
```diff
@@ -44,54 +44,150 @@ func setMemoryArea(ctrl controller, area image.Rectangle) {
...
+	// Amount by which buffer contents must be moved to align with the physical
+	// top-left corner of the display.
+	//
+	// TODO: The offset shifts the buffer contents to be aligned such that the
+	// translated position of the physical, on-display (0,0) location is at
+	// a multiple of 8 on the equivalent to the physical X axis. With a bit of
+	// additional work transfers for the TopRight and BottomLeft origins should
+	// not require per-pixel processing by exploiting image1bit.VerticalLSB's
+	// underlying pixel storage format.
+	BufferDstOffset image.Point
```
Comment (maruel) [inline waveshare2in13v2/drawing.go:64]: If you don't intent to export the struct, it'd be good to not export its fields.

File: waveshare2in13v2/waveshare213v2.go [hunk 2]
```diff
@@ -77,19 +77,30 @@ type Dev struct {
...
 	busy gpio.PinIn
 
+	bounds image.Rectangle
 	buffer *image1bit.VerticalLSB
 	mode   PartialUpdate
 
 	opts *Opts
 }
 
+type Corner uint8
```
Comment (maruel) [inline waveshare2in13v2/waveshare213v2.go:87]: Please document.

File: waveshare2in13v2/drawing.go (hunk 1 above)
Comment (hansmi) [reply waveshare2in13v2/drawing.go]: Done.

File: waveshare2in13v2/waveshare213v2.go (hunk 2 above)
Comment (hansmi) [reply waveshare2in13v2/waveshare213v2.go:89]: Done.

**Review by maruel — APPROVED** [review]

**Pull request merged**